### PR TITLE
dotCMS/core#21823 Allow PP filters to be added/updated via the UI or a REST call

### LINKED ISSUE #21823: Allow PP filters to be added/updated via the UI or a REST call
**Is your feature request related to a problem? Please describe.**
Currently the only way to list PP filters is by directly playing the updated YML files into the PP filter directory. This adds unnecessary complexity to updating these filters, especially for cloud customers as they need to engage support for updates to filters.

**Describe the solution you'd like**
While having a UI that allows users to create/configure PP filters directly would be ideal,  it is a complex thing to add.

Allow users to perform CRUD operations on the YML files, either via the backend or via a REST call. Also allow them to download their current configuration, as they need to know what they are working with.

The following GoogleDoc created by Jonathan: https://docs.google.com/document/d/1sbrX3711dsQe2t1U5snnuFpqMIQ3jaxiSmKEwrJZT1Q has all the details for every action that can be executed on Push Publishing Filters.

### PR DESCRIPTION
**Pull request by jdotcms**
Full crud for PP filters

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publishing/PublisherAPI.java [hunk 1]
```diff
@@ -58,6 +58,12 @@
      */
     List<FilterDescriptor> getFiltersDescriptorsByRole(final User user) throws DotDataException;
 
+    /**
+     * Returns true if the filter descriptor exists
+     * @return boolean
+     */
+    boolean existsFilterDescriptor(final String filterKey);
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/publishing/PublisherAPI.java:65]: Probably it's good idea to rename it to `filterDescriptorExists`

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

File: dotCMS/src/main/java/com/dotcms/publishing/PublisherAPI.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/publishing/PublisherAPI.java:65]: Agree

**Review by dsilvam — APPROVED** [review]

**Review by nollymar — CHANGES_REQUESTED** [review]
MSSQL tests are not running, please take a look: https://raw.githack.com/dotCMS/test-results/issue-21823-crud-ppfilters/projects/core/current/integration/mssql/reports/html/index.html

**Review by nollymar — APPROVED** [review]

**Pull request merged**